src/inmanta/protocol/endpoints.py

```diff
@@ -191,11 +209,13 @@ async def start(self) -> None:
         """
         assert self._env_id is not None
         LOGGER.log(3, "Starting agent for %s", str(self.sessionid))
-        self._client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout)
+        self._client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout, force_instance=True)
+        self._heartbeat_client = SessionClient(self.name, self.sessionid, timeout=self.server_timeout, force_instance=True)
         await self.start_connected()
         self.add_background_task(self.perform_heartbeat())
 
     async def stop(self) -> None:
+        self._heartbeat_client.close()
```

Comment:
Why do we close this one and not the other one?

Reply:
it should be closed as well. the docs say you need to close it yourself if you use force_instance. I first hadn't used it for the _client one and forgot to add the close() for that one afterwards.